calculator.c

```diff
@@ -0,0 +1,73 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <string.h>
+
+int precedence(char op) {
+    return (op == '*' || op == '/') ? 2 : (op == '+' || op == '-') ? 1 : 0;
+}
+
+int perform(int a, int b, char op, int *error) {
+    if (op == '/' && b == 0) {
+        *error = 1; 
+        return 0;
+    }
+    switch (op) {
+    case '+': return a + b;
+    case '-': return a - b;
+    case '*': return a * b;
+    case '/': return a / b;
+    default: return 0; 
+}
+
+}
+int evaluate(const char *exp,int *result){
```

Comment:
This function can be divided into small functions which do one specific task. Practice writing functions in such a way that one function does one task.

Reply:
Sure!!